src/nerpa_pipeline/handle_rban.py

```diff
@@ -154,76 +221,43 @@ def add_hybrid_monomers(graph):
+    if not backbones:
         log.warning(f'Structure "{rban_record["id"]}": unable to determine backbone sequence. '
                     f'Skipping "{rban_record["id"]}".')
-        return []
+        return graph_record, []
+
+    perm_backbones = permuted_backbones(backbones)
+
+    # all fragments individually
+    variants = [NRP_Variant(nrp_id=rban_record['id'],
+                            fragments=[backbone_to_fragment(backbone)])
+                for backbone in chain(backbones, perm_backbones)]
 
-    def gen_nerpa_input(path, cyclic=False):
-        inp_nodes = []
-        for node in path:
-            attr = G.nodes[node]
-            name = attr['name']
-            # if not attr.get('isIdentified', False):
-            #     name = f'<{name}>'
-            is_d = attr.get('isD', None)
-            if is_d is not None:
-                name = f'@D-{name}' if is_d else f'@L-{name}'
-            if attr.get('isPKHybrid', False):
-                name = f'*{name}'
-            inp_nodes.append(name)
-        inp_edges = [f'{i},{i+1}' for i in range(len(inp_nodes)-1)]
-        if cyclic:
-            inp_edges.append(f'{len(inp_nodes)-1},{0}')
-        res = ','.join(inp_nodes)
-        res += ';'
-        res += ';'.join(inp_edges)
-        return res
-
-    def sufficiently_covered(path):
-        n = sum(G.nodes[node]['isIdentified'] for node in path)
-        return n >= min_recognized_nodes
-
-    '''create a dictionary with records of all fragments
-    structures.keys() are the str records of fragments in Nerpa format
-    structures.values() are these fragments in the original graph (as List[int])'''
-    structures = {}
-    for path in cycles:
-        if sufficiently_covered(path):  # path/cycle contains enough recognized monomers
-            gr = gen_nerpa_input(path, cyclic=True)
-            structures[gr] = path
-    for path in paths:
-        if sufficiently_covered(path):
-            gr = gen_nerpa_input(path)
-            structures[gr] = path
-    if 1 < len(paths) < 4:
-        # try to join linear fragments in all possible ways
-        for x in permutations(paths):
-            path = [node for comp in x for node in comp]
-            if sufficiently_covered(path):  # kind of strange to check it inside the cycle, as 'coverage' does not depend on the permutation, but here we are
-                gr = gen_nerpa_input(path)
-                structures[gr] = path
-    structures = [(f'{structure_id}_variant{i}', gr, structures[gr]) for i, gr in enumerate(sorted(structures.keys()))]
-
-    return structures
+    # all fragments together
+    if len(backbones) > 1:
+        variants.append(NRP_Variant(nrp_id=rban_record['id'],
+                                    fragments=list(map(backbone_to_fragment, backbones))))
+
+    return graph_record, variants
 
 
 def rban_postprocessing(path_to_rban_output, main_out_dir, path_to_rban, path_to_monomers_db, log):
```

Comment:
I didn't notice in the prev pr: "This function tries to remove a lipid tail" should read as "This function tries to remove a polyketide chain"